apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/ir/BackendIrBuilder.kt

```diff
@@ -754,4 +756,15 @@ internal class BackendIrBuilder constructor(
       val condition: BackendIr.Condition,
       val selectionKeys: Set<SelectionKey>,
   )
+
+  private val FrontendIr.Selection.Field.normalizedTypeName: String
+    get() {
+      val isListType = if (this.type is FrontendIr.Type.NonNull) {
+        this.type.ofType is FrontendIr.Type.List
+      } else {
+        this.type is FrontendIr.Type.List
+      }
+      val isObjectType = this.selections.isNotEmpty()
+      return if (isListType && isObjectType) this.responseName.singularize().capitalize() else this.responseName
```

Comment:
Why `capitalize()` in one branch and not the other ?

Reply:
will fix this.